src/main/java/jenkins/plugins/logstash/configuration/ElasticSearch.java

```diff
@@ -125,33 +129,48 @@ public int getDefaultPort()
       return 9300;
     }
 
-    @Override
-    public FormValidation doCheckHost(@QueryParameter("value") String value)
+    public FormValidation doCheckUri(@QueryParameter("value") String value)
     {
       if (StringUtils.isBlank(value))
       {
         return FormValidation.warning(Messages.PleaseProvideHost());
       }
       try
       {
-        new URL(value);
+        URI uri = new URI(value);
+        validateUri(uri);
       }
-      catch (MalformedURLException e)
+      catch (URISyntaxException | IllegalArgumentException e)
       {
         return FormValidation.error(e.getMessage());
       }
       return FormValidation.ok();
     }
+  }
 
-    public FormValidation doCheckKey(@QueryParameter("value") String value)
-    {
-      if (StringUtils.isBlank(value))
+  /**
+   * Validates that the given uri has a scheme, a port and a path which is not empty or just "/"
+   *
+   * @param uri
+   * @throws IllegalArgumentException when one of scheme, port or path
+   */
+  public static void validateUri(URI uri) throws IllegalArgumentException
+  {
+      try
       {
-        return FormValidation.error(Messages.ValueIsRequired());
+        uri.toURL();
+      }
+      catch (MalformedURLException e)
+      {
+        throw new IllegalArgumentException(e.getMessage());
       }
 
-      return FormValidation.ok();
+    if(uri.getPort() == -1) {
```

Comment:
Why not let users rely on default port?
`https://logstash.internal/index` looks better than `https://logstash.internal:443/index`